Makefile.am

```diff
@@ -2683,6 +2684,23 @@ test_sdap_access_LDADD = \
     libdlopen_test_providers.la \
     $(NULL)
 
+test_sdap_certmap_SOURCES = \
+    src/tests/cmocka/test_sdap_certmap.c \
+    src/providers/ldap/sdap_certmap.c \
+    $(NULL)
+test_sdap_certmap_CFLAGS = \
+    $(AM_CFLAGS) \
+    $(TALLOC_CFLAGS) \
+    $(POPT_CFLAGS) \
+    $(NULL)
+test_sdap_certmap_LDADD = \
+    $(CMOCKA_LIBS) \
+    $(TALLOC_LIBS) \
+    $(POPT_LIBS) \
+    $(SSSD_INTERNAL_LTLIBS) \
+    libsss_test_common.la \
+    $(NULL)
```

Comment:
Linked failed with error
```
/usr/bin/ld: src/providers/ldap/test_sdap_certmap-sdap_certmap.o: undefined reference to symbol 'sss_certmap_init@@SSS_CERTMAP_0.0'
//dev/shm/gcc/.libs/libsss_certmap.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

You need to link with `libsss_certmap` or use a little bit different way:

```
diff --git a/Makefile.am b/Makefile.am
index 54f43a25b..57ad892a2 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -2686,7 +2686,6 @@ test_sdap_access_LDADD = \
 
 test_sdap_certmap_SOURCES = \
     src/tests/cmocka/test_sdap_certmap.c \
-    src/providers/ldap/sdap_certmap.c \
     $(NULL)
 test_sdap_certmap_CFLAGS = \
     $(AM_CFLAGS) \
@@ -2699,6 +2698,8 @@ test_sdap_certmap_LDADD = \
     $(POPT_LIBS) \
     $(SSSD_INTERNAL_LTLIBS) \
     libsss_test_common.la \
+    libsss_ldap_common.la \
+    libdlopen_test_providers.la \
     $(NULL)
```